services/libs/aster-std/src/syscall/mmap.rs

```diff
@@ -130,9 +133,26 @@ fn mmap_filebacked_vmo(
     };
     let map_addr = vm_map_options.build()?;
     trace!("map range = 0x{:x} - 0x{:x}", map_addr, map_addr + len);
+
+    check_mapped_range(map_addr, len, &option.flags);
+
     Ok(map_addr)
 }
 
+fn check_mapped_range(map_addr: Vaddr, len: usize, flags: &MMapFlags) {
+    if !flags.contains(MMapFlags::MAP_32BIT) {
+        return;
+    }
+
+    // MAP_32BIT requires the mapped range is below 2GB. This flag does not make much
+    // sense for 64bit systems. So we only check the range and report warnings if the
+    // mapped range is not below 2GB.
```

Comment:
Saying "This flag does not make much sense for 64bit systems" is misleading. Quite the opposite. The man page of `mmap` states that "this flag is supported _only on x86-64_, for 64-bit programs."

The man page continues, "Modern x86-64 processors no longer have this performance problem, so use of this flag is not required on those systems". So the more appropriate understanding should be that this flag is still supported by Linux, it is just that using it won't bring any benefit.

The man page further states that "The MAP_32BIT flag is ignored when MAP_FIXED is set".

For the reasons above, what should be done is to
1. Recognize this flag by modifying `MMapFlags` (as done by this PR);
2. If `flags.contains(MAP_FIXED)`, ignore (or remove) the `MAP_32BIT` bit (not done by this PR).
3. Otherwise, ensure that the mapping is created below 2GB (not done by this PR).

If mapping below 2GB cannot be implemented easily, then this PR should add a TODO and log a warning message, emphasizing that silently ignoring the flag is incorrect and should be fixed in the future.